Modules/Messages/src/Messages.Infrastructure/Persistence/Database/Repository/MessagesRepository.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Backbone.Modules.Messages.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Messages.Application.Extensions;
+using Backbone.Modules.Messages.Domain.Entities;
+using Backbone.Modules.Messages.Domain.Ids;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.Persistence.Database;
+using Enmeshed.BuildingBlocks.Application.Pagination;
+using Microsoft.EntityFrameworkCore;
+using Enmeshed.BuildingBlocks.Application.Extensions;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.UserContext;
+using Backbone.Modules.Messages.Application.Messages.Commands.SendMessage;
+using Enmeshed.DevelopmentKit.Identity.ValueObjects;
+using Backbone.Modules.Messages.Application.Messages.Queries.ListMessages;
+
+namespace Backbone.Modules.Messages.Infrastructure.Persistence.Database.Repository;
+public class MessagesRepository : IMessagesRepository
+{
+    private readonly DbSet<Message> _messages;
+    private readonly IQueryable<Message> _readOnlyMessages;
+    private readonly IUserContext _userContext;
+
+    public MessagesRepository(MessagesDbContext dbContext, IUserContext userContext)
```

Comment:
I'm not 100% convinced, but I think we shouldn't use the `IUserContext` in the repository. Instead, we should pass the active Identity into each method that needs it.
It's the responsibility of the Application layer to deal with users, and not the one of the Infrastructure layer.

Reply:
Agreed